daemonize/src/lib.rs

```diff
@@ -354,7 +354,7 @@ impl<T> Daemonize<T> {
                 Ok(Some(first_child_pid)) => {
                     Outcome::Parent(match waitpid(first_child_pid) {
                         Err(err) => Err(err.into()),
-                        Ok(first_child_exit_code) => Ok(Parent { first_child_exit_code: first_child_exit_code as i32 }),
+                        Ok(first_child_exit_code) => Ok(Parent { first_child_exit_code }),
```

Comment:
`waitpid` returns `libc::c_int`, which isn't necessarily an i32 on all platforms, hence the conversion.

(currently only for `target_arch` being `avr` or `msp430`, where it's an `i16`)

Reply:
Actually, correction, I was looking at `core::ffi::c_int`. Looks like currently all platforms the `libc` crate supports define it as `i32`. Still, it could theoretically be `i16` for some future platform.

Reply:
I have added back the integer cast, as well as a comment describing its purpose and a `#[allow(clippy::unnecessary_cast)]` macro to remove the warning. There are also two similar cases in `daemonize-tests` but I have not modified or commented on them in the code. The locations are both in daemonize-tests/src/lib.rs, at lines 184 and 185.

Reply:
Also, I believe the `waitpid` function returns a `pid_t` rather than a `c_int`. I couldn't find any case where `pid_t` isn't type aliased to a `i32` in the `libc` crate.